account_fiscal_year_closing/migrations/16.0.1.0.0/pre-migration.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2023 Giuseppe Borruso <gborruso@dinamicheaziendali.it>
+# @author: Giuseppe Borruso <gborruso@dinamicheaziendali.it>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openupgradelib import openupgrade
+
+
+def update_account_type(env, model, new_type, old_type):
+    openupgrade.logged_query(
+        env.cr,
+        """
+            UPDATE %(model)s
+            SET account_type = '%(new_type)s'
+            WHERE id in (
+                SELECT id
+                WHERE account_type_id = %(old_type)s
+            )
+        """
+        % {
+            "model": model,
+            "new_type": new_type,
+            "old_type": old_type,
+        },
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+
+    all_account_type = [
+        ("asset_receivable", env.ref("account.data_account_type_receivable").id),
```

Comment:
When migrating from v15 to v16 using OpenUpgrade, the following error occurs:
```
2024-07-30 09:16:17,113 2133688 INFO cbx_2024-07-30_v16 odoo.modules.loading: Loading module account_fiscal_year_closing (102/170)
2024-07-30 09:16:17,430 2133688 INFO cbx_2024-07-30_v16 odoo.modules.migration: module account_fiscal_year_closing: Running migration [>16.0.1.0.0] pre-migration
2024-07-30 09:16:17,444 2133688 INFO cbx_2024-07-30_v16 OpenUpgrade: account_fiscal_year_closing: pre-migration script called with version 15.0.1.0.0
2024-07-30 09:16:17,444 2133688 ERROR cbx_2024-07-30_v16 OpenUpgrade: account_fiscal_year_closing: error in migration script /opt/odoo/local/addons-oca-community/addons/account_fiscal_year_closing/migrations/16.0.1.0.0/pre-migration.py: 'account.account.type'
2024-07-30 09:16:17,444 2133688 ERROR cbx_2024-07-30_v16 OpenUpgrade: 'account.account.type'
Traceback (most recent call last):
  File "/srv/odoo/venv/lib/python3.10/site-packages/openupgradelib/openupgrade.py", line 2292, in wrapped_function
    func(
  File "/opt/odoo/local/addons-oca-community/addons/account_fiscal_year_closing/migrations/16.0.1.0.0/pre-migration.py", line 31, in migrate
    ("asset_receivable", env.ref("account.data_account_type_receivable").id),
  File "/opt/odoo/src/odoo/odoo/api.py", line 600, in ref
    record = self[res_model].browse(res_id)
  File "/opt/odoo/local/OpenUpgrade/openupgrade_framework/odoo_patch/odoo/api.py", line 39, in __getitem__
    return Environment.__getitem__._original_method(self, model_name)
  File "/opt/odoo/src/odoo/odoo/api.py", line 550, in __getitem__
    return self.registry[model_name](self, (), ())
  File "/opt/odoo/src/odoo/odoo/modules/registry.py", line 190, in __getitem__
    return self.models[model_name]
KeyError: 'account.account.type'
2024-07-30 09:16:17,500 2133688 WARNING cbx_2024-07-30_v16 odoo.modules.loading: Transient module states were reset
2024-07-30 09:16:17,501 2133688 ERROR cbx_2024-07-30_v16 odoo.modules.registry: Failed to load registry
Traceback (most recent call last):
  File "/opt/odoo/src/odoo/odoo/modules/registry.py", line 90, in new
    odoo.modules.load_modules(registry, force_demo, status, update_module)
  File "/opt/odoo/src/odoo/odoo/modules/loading.py", line 484, in load_modules
    processed_modules += load_marked_modules(cr, graph,
  File "/opt/odoo/src/odoo/odoo/modules/loading.py", line 372, in load_marked_modules
    loaded, processed = load_module_graph(
  File "/opt/odoo/src/odoo/odoo/modules/loading.py", line 183, in load_module_graph
    migrations.migrate_module(package, 'pre')
  File "/opt/odoo/local/OpenUpgrade/openupgrade_framework/odoo_patch/odoo/modules/migration.py", line 18, in migrate_module
    MigrationManager.migrate_module._original_method(self, pkg, stage)
  File "/opt/odoo/src/odoo/odoo/modules/migration.py", line 189, in migrate_module
    migrate(self.cr, installed_version)
  File "/srv/odoo/venv/lib/python3.10/site-packages/openupgradelib/openupgrade.py", line 2292, in wrapped_function
    func(
  File "/opt/odoo/local/addons-oca-community/addons/account_fiscal_year_closing/migrations/16.0.1.0.0/pre-migration.py", line 31, in migrate
    ("asset_receivable", env.ref("account.data_account_type_receivable").id),
  File "/opt/odoo/src/odoo/odoo/api.py", line 600, in ref
    record = self[res_model].browse(res_id)
  File "/opt/odoo/local/OpenUpgrade/openupgrade_framework/odoo_patch/odoo/api.py", line 39, in __getitem__
    return Environment.__getitem__._original_method(self, model_name)
  File "/opt/odoo/src/odoo/odoo/api.py", line 550, in __getitem__
    return self.registry[model_name](self, (), ())
  File "/opt/odoo/src/odoo/odoo/modules/registry.py", line 190, in __getitem__
    return self.models[model_name]
KeyError: 'account.account.type'
2024-07-30 09:16:17,501 2133688 CRITICAL cbx_2024-07-30_v16 odoo.service.server: Failed to initialize database `cbx_2024-07-30_v16`.
Traceback (most recent call last):
  File "/opt/odoo/src/odoo/odoo/service/server.py", line 1310, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "<decorator-gen-16>", line 2, in new
  File "/opt/odoo/src/odoo/odoo/tools/func.py", line 87, in locked
    return func(inst, *args, **kwargs)
  File "/opt/odoo/src/odoo/odoo/modules/registry.py", line 90, in new
    odoo.modules.load_modules(registry, force_demo, status, update_module)
  File "/opt/odoo/src/odoo/odoo/modules/loading.py", line 484, in load_modules
    processed_modules += load_marked_modules(cr, graph,
  File "/opt/odoo/src/odoo/odoo/modules/loading.py", line 372, in load_marked_modules
    loaded, processed = load_module_graph(
  File "/opt/odoo/src/odoo/odoo/modules/loading.py", line 183, in load_module_graph
    migrations.migrate_module(package, 'pre')
  File "/opt/odoo/local/OpenUpgrade/openupgrade_framework/odoo_patch/odoo/modules/migration.py", line 18, in migrate_module
    MigrationManager.migrate_module._original_method(self, pkg, stage)
  File "/opt/odoo/src/odoo/odoo/modules/migration.py", line 189, in migrate_module
    migrate(self.cr, installed_version)
  File "/srv/odoo/venv/lib/python3.10/site-packages/openupgradelib/openupgrade.py", line 2292, in wrapped_function
    func(
  File "/opt/odoo/local/addons-oca-community/addons/account_fiscal_year_closing/migrations/16.0.1.0.0/pre-migration.py", line 31, in migrate
    ("asset_receivable", env.ref("account.data_account_type_receivable").id),
  File "/opt/odoo/src/odoo/odoo/api.py", line 600, in ref
    record = self[res_model].browse(res_id)
  File "/opt/odoo/local/OpenUpgrade/openupgrade_framework/odoo_patch/odoo/api.py", line 39, in __getitem__
    return Environment.__getitem__._original_method(self, model_name)
  File "/opt/odoo/src/odoo/odoo/api.py", line 550, in __getitem__
    return self.registry[model_name](self, (), ())
  File "/opt/odoo/src/odoo/odoo/modules/registry.py", line 190, in __getitem__
    return self.models[model_name]
KeyError: 'account.account.type'
```

We believe that there is a timing issue with this pre-migration script, because at this stage, those records are already deleted and therefore the keyerror occurs

Reply:
update pre-commit
can you check again?